spec/create_new_milestone_action_spec.rb

```diff
@@ -0,0 +1,160 @@
+    run_action_with(:need_appstore_submission, true)
+  end
+
+  it 'uses the most recent milestone date to calculate the due date and version of new milestone' do
+    comment = "Code freeze: November 18, 2022\nApp Store submission: December 02, 2022\nRelease: December 02, 2022\n"
+    allow(client).to receive(:list_milestones).and_return(milestone_list)
+    expect(client).to receive(:create_milestone).with(test_repository, '10.5', due_on: '2022-11-18T12:00:00Z', description: comment)
+    run_described_fastlane_action(default_params)
+  end
+
+  it 'raises an error when the due date of milestone does not exists' do
+    allow(client).to receive(:list_milestones).and_return([{ title: '10.1', number: '1234' }])
+    expect { run_described_fastlane_action(default_params) }.to raise_error(FastlaneCore::Interface::FastlaneError, 'Milestone 10.1 has no due date.')
+  end
+
+  it 'raises an error when the milestone is not found or does not exist on the repository' do
+    allow(client).to receive(:list_milestones).and_return([])
+    expect { run_described_fastlane_action(default_params) }.to raise_error(FastlaneCore::Interface::FastlaneError, 'No milestone found on the repository.')
+  end
+
+  describe 'initialize' do
+    context 'with github_token' do
+      it 'properly passes the environment variable `GITHUB_TOKEN` to Octokit::Client' do
+        ENV['GITHUB_TOKEN'] = test_token
+        expect(Octokit::Client).to receive(:new).with(access_token: test_token)
+        run_action_without_key(:github_token)
+      end
+
+      it 'properly passes the parameter `:github_token` to Octokit::Client' do
+        expect(Octokit::Client).to receive(:new).with(access_token: test_token)
+        run_described_fastlane_action(default_params)
+      end
+
+      it 'prioritizes `:github_token` parameter over `GITHUB_TOKEN` environment variable if both are present' do
+        ENV['GITHUB_TOKEN'] = 'Test-EnvGithubToken-1234'
+        expect(Octokit::Client).to receive(:new).with(access_token: test_token)
+        run_described_fastlane_action(default_params)
+      end
+    end
+
+    context 'when using default parameters' do
+      let(:github_helper) do
+        instance_double(
+          Fastlane::Helper::GithubHelper,
+          get_last_milestone: test_milestone,
+          create_milestone: nil
+        )
+      end
+
+      before do
+        allow(Fastlane::Helper::GithubHelper).to receive(:new).and_return(github_helper)
+      end
+
+      it 'uses default value when neither `GHHELPER_NUMBER_OF_DAYS_FROM_CODE_FREEZE_TO_RELEASE` environment variable nor parameter `:number_of_days_from_code_freeze_to_release` is present' do
+        default_code_freeze_days = 14
+        expect(github_helper).to receive(:create_milestone).with(
+          anything,
+          anything,
+          anything,
+          anything,
+          default_code_freeze_days,
+          anything
+        )
```

Comment:
I wonder if it wouldn't be the occasion to make this `GithubHelper#create_milestone` method start using named parameters instead of positional ones — that are even harder to identify in test cases like this amongst the other `anything` placeholders. That should then allow us to use `.with(hash_including(code_freeze_days: 14))` instead of having to use all those `anything` placeholders there.

But that means a small refactoring of all the call sites though (which might be outside of this PR's scope… but depending on how big a change it would be—not sure there are that many call sites, the `create_new_milestone` action might even be the only one?—, that could still be worth it so we can then have both the call sites and those unit tests be way nicer…
I'll let you evaluate how much changes it would warrant to do this, and decide if it's worth it or not.

Reply:
From what I see, the only places that will be affected are the `GithubSpec` and the `create_milestone_action/Spec`, I agree that this method should have the named params to "force" it to be better legible and cleaner. 

However, TBH I do not feel that I should do it on this PR, as it is a bit out of scope (although is a small change). So, I'll open another PR to deal with this, and update these tests here once this new PR is merged 😄

Reply:
Update: This is addressed on PR #426 😄